graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/context/GraphQLContextContributor.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.context
+
+import graphql.GraphQLContext
+import org.springframework.http.HttpHeaders
+import org.springframework.web.context.request.WebRequest
+
+interface GraphQLContextContributor {
+    fun contribute(
+        builder: GraphQLContext.Builder,
+        extensions: Map<String, Any>?,
+        headers: HttpHeaders?,
+        webRequest: WebRequest?
```

Comment:
Anything else we should make available via this signature?
ie:
-         variables: Map<String, Any>,
-         extensions: Map<String, Any>?,
-         operationName: String?,

Reply:
Aren't the headers available in the webrequest?

Reply:
Was just thinking that, that what I get from copying DgsCustomContextBuilderWithRequest ;)

Reply:
Hmm.... I tried but then hit friction when looking at DefaultDgsQueryExecutor.getJsonResult ... that passes in headers but null for the webRequest....

Reply:
https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultDgsQueryExecutor.kt#L154-L155

Reply:
Looks like I just need to add an additional signature executeAndExtractJsonPath on interface DgsQueryExecutor and its implementation

Reply:
Done.

Reply:
Anything else we should make available via this signature?
ie:

query: String
variables: Map<String, Any>
operationName: String?